Creating of databases and questions removed to the store.

Tomislav-Zoricic opened the pull request.

Tomislav-Zoricic committed:
Creating of databases and questions removed to the store.

b0rza commented on src/components/professor/questions/modals/AddQuestionModal.vue:
```diff
@@ -101,22 +102,25 @@ export default {
-      // NOTE Validate question adding.
-      const $addBtn = $(this.$refs.addQuestionButton)
+      const $addBtn = $(this.$refs.createQuestionButton)
       $addBtn.addClass('is-loading')
```
createQuestionButton should have the class automatically toggled added via a property binding.
Toggle that prop after an action finishes instead of touching the class manually.

b0rza commented on src/components/professor/databases/CreateDatabase.vue:
```diff
@@ -44,28 +45,24 @@ export default {
-        toastr.success('Creating database', 'Successful')
+    createDatabase () {
+      this.newDatabase(this.database)
+        .then(({ id }) => {
```
Make sure this `.then` and the `.catch` below are indented the same. Not sure if your linter should pick up on this?

b0rza commented on src/components/professor/questions/modals/AddQuestionModal.vue:
```diff
@@ -101,22 +102,25 @@ export default {
       $addBtn.addClass('is-loading')
 
-      axios.post('http://localhost:3000/questions', {
+      const q = {
```
`q` is a poor name for a variable, use `questionPayload`, or `questionMeta` or anything more prominent.

Also, bear in mind that some of these properties are going to have their names changed (eg. `question` becomes  `content` as per @zemmao's db blueprint). No need to change it within this PR, but make sure it's in one of the following ones.

b0rza commented on src/store/databases.js:
```diff
@@ -64,9 +67,43 @@ const actions = {
+
+  createDatabase ({ commit }, data) {
+    const messageSubtitle = 'Creating database'
+    return axios.post(`${URL}/databases`, data)
```
Which url, what if there are more? `API_URL` is a better name.

b0rza commented on src/store/databases.js:
```diff
@@ -64,9 +67,43 @@ const actions = {
 }
 
+// statusText added for later use.
+function checkStatus(status, statusText) {
```
👍  but rewrite `getQuestions` and `getDatabases` actions to use this helper as well.

Tomislav-Zoricic committed:
Made requested changes in PR

Changing variable names
Indenting
More Vue style property bindings
Cleaning up code

b0rza reviewed (approved):
🎉
